packages/server/src/api/routes/tests/application.spec.ts

```diff
@@ -324,4 +324,39 @@ describe("/applications", () => {
       expect(events.app.unpublished).toBeCalledTimes(1)
     })
   })
+
+  describe("app sync", () => {
+    it("should not sync automation logs", async () => {
+      // setup the apps
+      await config.createApp("testing-auto-logs")
+      const automation = await config.createAutomation()
+      await config.publish()
+      await context.doInAppContext(config.getProdAppId(), () => {
+        return config.createAutomationLog(automation)
+      })
+
+      // do the sync
+      const appId = config.getAppId()
+      await request
+        .post(`/api/applications/${appId}/sync`)
+        .set(config.defaultHeaders())
+        .expect("Content-Type", /json/)
+        .expect(200)
+
+      // does exist in prod
+      await context.doInAppContext(config.getProdAppId(), async () => {
+        const logs = await config.getAutomationLogs()
+        expect(logs.data.length).toBe(1)
+      })
+
+      // delete prod app so we revert to dev log search
+      await config.unpublish()
```

Comment:
This seems like pretty un-deterministic behaviour, is there not a way retrieve the desired logs by id?

Reply:
No - the issue isn't by ID, its by database, the way the system is designed it retrieves from prod then from dev. I can probably make this more clear in the test as the context shifts aren't really needed.

Reply:
By "id" I mean the app id vs the prod id - essentially the different databases. 

I think the logic is just unclear? Needing to remove the prod db in order to query the dev db logs seems counter intuitive. Often tests highlight things like this which can be an opportunity to improve the interface. 

NAB = just wondered if there was an easier way to get the dev db logs.

Reply:
Yeah - there isn't a particularly nice way to do this - without basically writing a bunch of code into the SDK to provide access to something that isn't supposed to be accessible.